### DISCUSSION TITLE
useToken for ERC721

### DISCUSSION THREAD

**Discussion by bayugtt** [wevm/wagmi#2832] [Q&A] [answered]
i try to use it for erc721 but it seem it can't.
is there any chance to retreive balance of NFTs using wagmi?

its great to have function to check NFTs of connected address.
Thank you WAGMI

**Comment by SaadNiaziDev** [comment]
useBalance works with NFTs (Only for those collection it will get balance whose contract address you will give in function call). Make sure to have correct chain and correct NFT contract address.

**Reply by bayugtt** [reply]
if you don't mind, could you give me an example of getting NFTs balance.

for me its getting error on decimals() when i try with nft contract. its working amazing with erc20.

**Reply by SaadNiaziDev** [reply, marked as answer]
`const { data, isLoading, isError, isSuccess } = useContractRead( {`
        `address: nftAddress,`
        `abi: erc721ABI,`
        `chainId:1,`
        `functionName: 'balanceOf',`
        `args: [ address ],`
    `} )`

place nft contract address in place of nftAddress and your or whose so ever address, you want to get balance of in place of address. Also specify chainId

**Reply by bayugtt** [reply]
i didn't think that way.
thank you so much
